lib/spack/spack/modules.py

```diff
@@ -134,6 +134,13 @@ def add_path(path_name, directory):
                 if site_packages:
                     add_path('PYTHONPATH', site_packages[0])
 
+            # Add perl path unless it's an actual perl installation
+            # TODO: is there a better way to do this?
```

Comment:
The Python one changed recently, so I'd look at what it does there. This might also be something for `setup_dependent_environment`.